demisto_sdk/commands/common/hook_validations/integration.py

```diff
@@ -599,6 +601,102 @@ def _get_invalid_dbot_outputs(context_outputs_paths, context_outputs_description
+
+        return result
+
+    def validate_spelling(self, command: dict) -> bool:
+        """Validates command ouputs listed MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES
+           are spelled and capitalized correctly.
+        Returns:
+            True if command outputs are spelled right.
+        """
+        result = True
+        for output in command.get("outputs") or []:
+            context_path = output.get("contextPath", "")
+
+            for reputation_object in MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES:
+                if reputation_object.lower() in context_path.lower():
+                    if (
+                        reputation_object not in context_path
+                    ):  # the output is not spelled as expected
+                        (
+                            error_message,
+                            error_code,
+                        ) = Errors.command_reputation_output_capitalization_incorrect(
+                            command.get("name"), context_path, reputation_object
+                        )
+                        if self.handle_error(
+                            error_message,
+                            error_code,
+                            file_path=self.file_path,
+                            warning=self.structure_validator.quiet_bc,
+                        ):
+                            result = False
+        return result
+
+    @error_codes("IN158,IN158")
+    def validate_command_outputs(self) -> bool:
+        """Check if a command is using the following objects: file, ip, endpoint, email, domain, url, cve,
+        infofile, certificate, if so, validate it is according to context
+        standards: https://xsoar.pan.dev/docs/integrations/context-standards
+        Returns:
+            bool.
+            Whether a command is using the mentioned objects and their outputs
+            according to the context standards
+        """
+        commands = self.current_file.get("script", {}).get("commands", [])
+        result = True
+        for command in commands:
+            reputation_objects: Set[str] = set()
+            context_outputs_paths: Set[str] = set()
+            for output in command.get("outputs") or []:
+                context_path = output.get("contextPath", "")
+                context_outputs_paths.add(context_path)
+                for reputation_object in MANDATORY_REPUTATION_CONTEXT_OBJECTS_NAMES:
+                    if reputation_object.lower() in context_path.lower():
+                        reputation_objects.add(reputation_object)
+            if not (
+                self.validate_spelling(command)
+                and self.validate_all_reputation_outputs_exist(
+                    command, reputation_objects, context_outputs_paths
+                )
+            ):
+                result = False
+        return result
+
     @error_codes("DB100,DB101,IN107")
     def is_outputs_for_reputations_commands_valid(self) -> bool:
```

Comment:
check if possible to validate reputation_outputs here as well. we already iterating over command